### PR TITLE
Fix GeneratedId failed to query By Id.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Add IT test for GeneratedId, both Vertex and Edge.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

**Comment by Incarnation-p-lee** [comment]
Most code change due to IT test. Bug fix change should be
`11 files changed, 105 insertions(+), 37 deletions(-)`

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/result/AbstractGremlinResultReader.java [hunk 1]
```diff
@@ -16,11 +16,12 @@
 import java.util.Map;
 
 @NoArgsConstructor
+// TODO: seems only for Vertex.
 public abstract class AbstractGremlinResultReader {
 
     /**
      * properties's organization is a little complicated.
-     *
+     * <p>
```
Comment (xscript) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/result/AbstractGremlinResultReader.java]: Is this a typo or by design?

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/result/AbstractGremlinResultReader.java]: Sorry typo.
